dataservice/extensions.py

```diff
@@ -2,6 +2,59 @@
 from flask_sqlalchemy import SQLAlchemy, Model
 
 
-db = SQLAlchemy()
+class CRUDMixin(Model):
+    """
+    Mixin that adds convenience methods for CRUD
+    (create, read, update, delete) operations.
+    """
+
+    @classmethod
+    def create(cls, **kwargs):
+        """
+        Create a new record and save it the database.
+        """
+        instance = cls(**kwargs)
+        return instance.save()
+
+    def update(self, commit=True, **kwargs):
+        """
+        Update specific fields of a record.
+        """
+        for attr, value in kwargs.items():
+            setattr(self, attr, value)
```

Comment:
We'll want some logic in here to make sure certain properties aren't illegally modified, ie `date_created`.

I've been thinking of how this could be added by extending [here](https://github.com/noirbizarre/flask-restplus/blob/master/flask_restplus/resource.py#L60)

Reply:
I think there are various ways to address validation. Not sure what we want to do yet, but here are some options:

1. Do all validation in the web layer (in http methods) using the namespace models/serializers (or later if we use marshmallow, we would use marshmallow schemas). For example if our request model does not include the `date_created` key, then that attribute will not be in kwargs, and will never be illegally modified in the db. Also, if we have common serializers (like you implemented already and used in person entity) then we can enforce this behavior for all our entities. Then presumably kwargs is a set of valid parameters to execute the db operation. 

2. Do validation in both the web layer and the db layer. So do the above and maybe also modify CRUD Mixin to handle exceptions appropriately.

I guess we need to discuss more..

Reply:
One more thing, it looks like the namespace expect decorator can take a `validate` kwarg. So we can check for existence of required fields, etc.